### PR TITLE
Lab BOM based on lab spending guidelines (first draft)

### PR DESCRIPTION
**Pull request by mgheorghe**
a stab at creating a BOM based on https://github.com/opiproject/opi-poc/pull/515
looking for feedback on:

- BOM columns
- budget
- list of DPU/IPU that will support OPI
- should each _PU have it's own server or put as many as possible in a server (cost per _PU)
- 1, 3 or 5 year warranty / support ?
- any other feedback

### CODE REVIEW

**Comment by pdp2shirts** [comment]
On the 3-5 year support issue. In previous life managing Labs, I would always opt to not get 24X7 4-hour replacement support, and get the lowest level of support the vendor offered which was generally a 12x5 (business hours) and Next Business day part replacement. The cost savings was significant and Lab equipment almost never needs critical business system level of support. and with this being in a shared lab, we won't likely have 24x7 4 hours response time available from the Lab administrator team.

**Comment by pdp2shirts** [comment]
On the 1 DPU per server vs 1 Server and 8 DPUs. I would say we go with 1:1 to begin with. and have a test case that also has a single host with multiple dpus to cover the complexities of provisioning that brings. But let's start with the easiest case of deploying one dpi per server.

**Comment by kkongusa** [comment]
Has anyone looked at https://opendcim.org/ to document BOM + topology configuration details?   If all the information required by opendcim is captured in the *.md, that gives you confidence there are no gaps as far as information content is concerned.  Just a thought.

**Comment by mgheorghe** [comment]
> Has anyone looked at https://opendcim.org/ to document BOM + topology configuration details? If all the information required by opendcim is captured in the *.md, that gives you confidence there are no gaps as far as information content is concerned. Just a thought.

I will have a look and revise the pr if i missed anything

**Comment by kkongusa** [comment]
fyi, I created a PR to check in spending_guidelines.md in the official Policies folder of the opi repo: https://github.com/opiproject/opi/pull/45

**Comment by seroyer** [comment]
@mgheorghe #558 was merged which fixes the YAMLlint issue.  If you push a change or rebase, that should fix it.

File: lab/bom.md [hunk 1]
```diff
@@ -0,0 +1,164 @@
...
+| DPU Host (1 or 2 _PUs)           | Supermicro   | RAX XS4-11S3-10G   | 10       |          | $5, 131  | $51, 310  | [Shop](https://www.thinkmate.com)                                                                                                                                                                                                                                   |                                                                                                                                                                                                                                                         |
+| Storage Host                     | ASUS         | ESC8000A-E11       | 1        |          | $14, 000 | $14, 000  | [Product](https://www.asus.com/commercial-servers-workstations/esc8000a-e11/)                                                                                                                                                                                          |
```
Comment (glimchb) [inline lab/bom.md:24]: I think we can skip storage host, just use software defined storage on regular server

**Comment by glimchb** [comment]
@mgheorghe seems like 43 commits, can you please git rebase instead of merge ?

**Comment by mgheorghe** [comment]
> @mgheorghe seems like 43 commits, can you please git rebase instead of merge ?
yes last night i tried to do the dco signoff and i messed something badly, working now on fixing it

**Comment by glimchb** [comment]
@mgheorghe you can squash all commits to one now will be easier to fix DCO issues

File: lab/bom.md (hunk 1 above)
Comment (mgheorghe) [reply lab/bom.md]: removed from phase1

**Comment by mgheorghe** [comment]
i set the diagram to be 100% with, but the way to do it triggers the linter failure
unable to find a github compatible md way of setting image width to 100% except by using html tags

File: lab/bom.md [hunk 2]
```diff
@@ -0,0 +1,218 @@
...
+
+another table probably needed to include the actual deployment info:
+
+* rack space for each device
+* power used by each device
+* power to heat to HVAC math
+
+## Phase 1 testbed diagram
+
+<img src="./opi-rack-phase1.svg" width="100%">
```
Comment (glimchb) [inline lab/bom.md:64]: just use
```
![xPU Rack phase 1](./opi-rack-phase1.svg)
```

**Comment by pdp2shirts** [comment]
@opiproject/opi-maintainers we reviewed this PR on lab call can this be merged and future changes will be made as needed.

**Comment by kkongusa** [comment]
> @opiproject/opi-maintainers we reviewed this PR on lab call can this be merged and future changes will be made as needed.

No need to merge this one, since the official one already merged into the main opi repo, here: https://github.com/opiproject/opi/blob/main/Policies/spending_guidelines.md

Am assuming there is no need to merge to opi-poc repo.  Please confirm, so I can close.

**Comment by mgheorghe** [comment]
> > @opiproject/opi-maintainers we reviewed this PR on lab call can this be merged and future changes will be made as needed.
> 
> No need to merge this one, since the official one already merged into the main opi repo, here: https://github.com/opiproject/opi/blob/main/Policies/spending_guidelines.md
> 
> Am assuming there is no need to merge to opi-poc repo. Please confirm, so I can close.

will be good to merge it in.

**Comment by kkongusa** [comment]
> > > @opiproject/opi-maintainers we reviewed this PR on lab call can this be merged and future changes will be made as needed.
> > 
> > 
> > No need to merge this one, since the official one already merged into the main opi repo, here: https://github.com/opiproject/opi/blob/main/Policies/spending_guidelines.md
> > Am assuming there is no need to merge to opi-poc repo. Please confirm, so I can close.
> 
> will be good to merge it in.

Ugh! Brain freeze.  Somehow I thought this PR was for spending_guidlines.md, which it is not.  Yeah, I agree this PR, which speaks to the BOM, needs to be merged in at least the opi-poc repo.

**Comment by glimchb** [comment]
> @opiproject/opi-maintainers we reviewed this PR on lab call can this be merged and future changes will be made as needed.

this PR fails DCO. please squash all commits and signoff to fix DCO issue

**Comment by kkongusa** [comment]
> > @opiproject/opi-maintainers we reviewed this PR on lab call can this be merged and future changes will be made as needed.
> 
> this PR fails DCO. please squash all commits and signoff to fix DCO issue

In case no one actually googled it, here are the detailed steps 8-)

Squashing a commit
In GitHub Desktop, click Current Branch.
In the list of branches, select the branch that has the commits that you want to squash.
Click History.
Select the commits to squash and drop them on the commit you want to combine them with. ...
Modify the commit message of your new commit. ...
Click Squash Commits.

**Comment by glimchb** [comment]
and don't forget to click signoff :-)

**Review by glimchb — APPROVED** [review]

**Pull request merged**